src/Estimators/MomentumDistribution.cpp

```diff
@@ -22,18 +22,16 @@ namespace qmcplusplus
 MomentumDistribution::MomentumDistribution(MomentumDistributionInput&& mdi,
                                            size_t np,
                                            const PosType& twist_in,
-                                           const Lattice& lattice_in,
+                                           Lattice lattice_in,
                                            DataLocality dl)
-    : OperatorEstBase(dl),
+    : OperatorEstBase(dl, mdi.get_name(), mdi.get_type()),
       input_(std::move(mdi)),
       twist(twist_in),
-      lattice(lattice_in),
+      lattice(std::move(lattice_in)),
```

Comment:
The old code only pays the price of a copy constructor.
The new code pays the price of both copy and move constructor. In addition, do we really have the move constructor of `Lattice` implemented and tested?
I feel the old code was better.

Reply:
See
https://xania.org/202101/cpp-by-value-args

Move's should be basically free compared to anything that causes an Allocation. But with our own objects there is a bunch of auditing to be done to be clear on that. And anytime something is an rvalue this can save an extra copy you'd have with a const &.

Lattice has the implicitly defined default move constructor.  Which means this is likely equivalent to the const  better in most cases, but really I just wanted to shut up clang tidy.  I think the thing to do for now is remove this clang-tidy check.

Reply:
I see where you are from. That blog explains the case when the caller pass a raw string (const char *) which is not our case. Most of the time our callers pass `std::string` from `get_type()` and thus "const std::string&" wins.
I still saw two occurrence of `Lattice lattice` that needs to be stored.